### PR TITLE
New Role management module

### PR DESCRIPTION
**Pull request by rjeffman**
There is a new role management module placed in the plugins folder:

    plugins/modules/iparole.py

The role module allows to ensure presence or absence of roles and
manage role members.

Here is the documentation for the module:

    README-role.md

New example playbooks have been added:

    playbooks/role/role-is-absent.yml
    playbooks/role/role-is-present.yml
    playbooks/role/role-member-group-absent.yml
    playbooks/role/role-member-group-present.yml
    playbooks/role/role-member-host-absent.yml
    playbooks/role/role-member-host-present.yml
    playbooks/role/role-member-hostgroup-absent.yml
    playbooks/role/role-member-hostgroup-present.yml
    playbooks/role/role-member-privilege-absent.yml
    playbooks/role/role-member-privilege-present.yml
    playbooks/role/role-member-service-absent.yml
    playbooks/role/role-member-service-present.yml
    playbooks/role/role-member-user-absent.yml
    playbooks/role/role-member-user-present.yml
    playbooks/role/role-members-absent.yml
    playbooks/role/role-members-present.yml
    playbooks/role/role-rename.yml

New tests for the module:

    tests/role/test_role.yml
    tests/role/test_role_service_member.yml

### CODE REVIEW

File: README-role.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+
+Requirements
+------------
+
+**Controller**
+* Ansible version: 2.8+
+
+**Node**
+* Supported FReeIPA version (see above)
```
Comment (dawud) [inline README-role.md:31]: Typo, should be FreeIPA

Comment (rjeffman) [reply README-role.md:31]: Thank you.

File: tests/role/env_facts.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+- name: Get Domain from server name
+  set_fact:
+    ipaserver_domain: "{{ groups.ipaserver[0].split('.')[1:] | join ('.') }}"
```
Comment (seocam) [inline tests/role/env_facts.yml:4]: Can we use `ansible_fqdn` here?

File: tests/role/test_role.yml [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+      ipaadmin_password: SomeADMINpassword
+      name: testrole
+      privilege:
+      - Host Administrators
+      action: member
+      state: absent
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure role has member has privileges restored.
```
Comment (seocam) [inline tests/role/test_role.yml:91]: Missing idempotency check for this test.

Comment (rjeffman) [reply tests/role/test_role.yml:91]: Fixed.

File: tests/role/env_facts.yml (hunk 2 above)
Comment (rjeffman) [reply tests/role/env_facts.yml:4]: yes, we can. fixed.

**Pull request merged**